sql/core/src/test/scala/org/apache/spark/sql/execution/metric/SQLMetricsSuite.scala

```diff
@@ -163,29 +178,38 @@ class SQLMetricsSuite extends SparkFunSuite with SharedSQLContext {
     val df2 = testData2.groupBy('a).count()
     val expected2 = Seq(
       Map("number of output rows" -> 4L,
-        "avg hashmap probe (min, med, max)" -> "\n(1, 1, 1)"),
+        "avg hash probe (min, med, max)" -> "\n(1, 1, 1)"),
       Map("number of output rows" -> 3L,
-        "avg hashmap probe (min, med, max)" -> "\n(1, 1, 1)"))
+        "avg hash probe (min, med, max)" -> "\n(1, 1, 1)"))
     testSparkPlanMetrics(df2, 1, Map(
       2L -> ("HashAggregate", expected2(0)),
       0L -> ("HashAggregate", expected2(1)))
     )
   }
 
   test("Aggregate metrics: track avg probe") {
-    val random = new Random()
-    val manyBytes = (0 until 65535).map { _ =>
-      val byteArrSize = random.nextInt(100)
-      val bytes = new Array[Byte](byteArrSize)
-      random.nextBytes(bytes)
-      (bytes, random.nextInt(100))
-    }
-    val df = manyBytes.toSeq.toDF("a", "b").repartition(1).groupBy('a).count()
-    val metrics = getSparkPlanMetrics(df, 1, Set(2L, 0L)).get
-    Seq(metrics(2L)._2("avg hashmap probe (min, med, max)"),
-        metrics(0L)._2("avg hashmap probe (min, med, max)")).foreach { probes =>
-      probes.toString.stripPrefix("\n(").stripSuffix(")").split(", ").foreach { probe =>
-        assert(probe.toInt > 1)
+    // Assume the execution plan is:
+    // Wholestage disabled:
+    // LocalTableScan(nodeId = 4) ->Exchange (nodeId = 3) -> HashAggregate(nodeId = 2) ->
+    // Exchange(nodeId = 1) -> HashAggregate(nodeId = 0)
```

Comment:
can we use tree string like https://github.com/apache/spark/pull/18301/files#diff-117230dd7763cbc4f5da0be2f3a948ecR300